### PR TITLE
[DEV-6618] Agency Recipients endpoint

### PR DESCRIPTION
**Pull request by emily-brents**
**Description:**
Creates `/api/v2/agency/{toptier_code}/recipients/` endpoint that returns the data points required for the box-and-whiskers chart on the Agency V2 Profile page

**Technical details:**
Uses the `percentile_disc` postgres function to calculate the percentiles, also uses normal max, min, count, etc functions

**Requirements for PR merge:**

1. [X] Unit & integration tests updated
2. [X] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
    - [x] Frontend 
4. [X] Matview impact assessment completed
5. [X] Frontend impact assessment completed
6. [X] Data validation completed
7. [N/A] Appropriate Operations ticket(s) created
8. [X] Jira Ticket [DEV-6618](https://federal-spending-transparency.atlassian.net/browse/DEV-6618):
    - [X] Link to this Pull-Request
    - [X] Performance evaluation of affected (API | Script | Download)
    - [X] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

**Review by collinwr — APPROVED** [review]
LGTM. Approved

File: usaspending_api/api_contracts/contracts/v2/agency/toptier_code/recipients.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+FORMAT: 1A
+HOST: https://api.usaspending.gov
+
+# Recipients [/api/v2/agency/{toptier_code}/recipients/{?fiscal_year}]
+
+Returns a list of data points of an agencies recipients for a given fiscal year
+
+## GET
+
++ Request (application/json)
```
Comment (ebdabbs) [inline usaspending_api/api_contracts/contracts/v2/agency/toptier_code/recipients.md:10]: Lines  10-16 are unnecessary for a GET request and prevent the mock API from working
```
[MATCHING] by url pattern: /api/v2/agency/:toptier_code/recipients/ MATCHED
[MATCHING] by request content type: application/json actual:  NOT_MATCHED
```

**Review by ebdabbs — CHANGES_REQUESTED** [review]
One minor change request! This looks good for the original AC / design, but I think we'll need to add to this endpoint and have a discussion with UX about the latest mockup

File: usaspending_api/api_contracts/contracts/v2/agency/toptier_code/recipients.md (hunk 1 above)
Comment (emily-brents) [reply usaspending_api/api_contracts/contracts/v2/agency/toptier_code/recipients.md]: Okay, should be good to go.

**Review by ebdabbs — COMMENTED** [review]
API contract approved

**Review by ebdabbs — APPROVED** [review]

File: usaspending_api/agency/v2/views/recipients.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            .distinct()
+            .count(),
+            "max": results["max"],
+            "min": results["min"],
+            "25th_percentile": results["25th_percentile"],
+            "50th_percentile": results["50th_percentile"],
+            "75th_percentile": results["75th_percentile"],
+            "messages": self.standard_response_messages,
+        }
+        return results
```
Comment (sethstoudenmier) [inline usaspending_api/agency/v2/views/recipients.py:59]: There are two `return` statements here. The second will never be reached.

**Review by ebdabbs — APPROVED** [review]

**Review by collinwr — APPROVED** [review]

**Pull request merged**